sg.config.yaml

```
@@ -59,6 +59,7 @@ env:
       { "Name": "embeddings", "Host": "127.0.0.1:6099" },
       { "Name": "zoekt-index-0", "Host": "127.0.0.1:6072" },
       { "Name": "zoekt-index-1", "Host": "127.0.0.1:6073" },
+      { "Name": "syntactic-codeintel-worker", "Host": "127.0.0.1:6075" },
```

We will want this to be horizontally scalable (I think we should have this be part of the Q2 design goals).

So maybe we should have the default configuration with two replicas? That would help catch issues where we accidentally design/implement stuff without taking horizontal scaling into account.

```suggestion
      { "Name": "syntactic-code-intel-worker-0", "Host": "127.0.0.1:6075" },
      { "Name": "syntactic-code-intel-worker-1", "Host": "127.0.0.1:6076" },
```

---

I agree, but IMO for MVP we should keep things simple. Especially as this will be part of Q2 - there we can focus on making this scalable, at the moment it'd be more of a distraction IMO, as we need to verify the core parts of this worker first.

---

My bad, I misspoke, I was thinking it would be part of the _Q1_ design goals. Not necessarily to arbitrary numbers of workers, but at least a handful, similar to Zoekt/gitserver.

So slightly pushing back.

If it's adding significant complexity to the implementation, we can change this back. But from the point of view of avoiding having to rework things, it'd be valuable to have this catch things like making sure we're propagating the worker identity consistently when work is done (instead of implicitly coding against a simpler configuration with a single worker).

---

Of the complexities I can imagine right now:

1. Load balancing API (no idea how it's currently done for services that are triggered from public API)
2. Sharding repo cache - we will probably neet some consistent hashing to make sure workers can scale up and down and not reclone repositories that are handled by other instances

For now, I'll add the two instances to the local configuration and we'll see if it causes any significant complexity.

---

Done in https://github.com/sourcegraph/sourcegraph/pull/59747/commits/29345383fa477488e0a0373b5ce787b1e4771fe9

---

For those aspects, we can have a simpler/no-op implementation for the first pass.
- Hard-code number of workers (no automatic scaling).
- No extra load-balancing/redundancy.
- Decide repo simply based on a simple `hash(repo_uri) % N` instead of consistent hashing.